### PR TITLE
Add TaikoJudgementInfo.

### PR DESCRIPTION
**Pull request by smoogipoo**

### CODE REVIEW

File: osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs [hunk 1]
```diff
@@ -7,5 +7,76 @@ namespace osu.Game.Modes.Taiko.Judgements
...
     {
+        /// <summary>
+        /// The maximum score value.
+        /// </summary>
+        public const TaikoScoreResult MAX_SCORE = TaikoScoreResult.Great;
+
+        /// <summary>
+        /// The score value.
+        /// </summary>
+        public TaikoScoreResult Score;
```
Comment (peppy) [inline osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs]: Weird having vairables called `Score` but enums being `ScoreResult`. I think we should call this the `Result` for now (`Score` is also used to represent a completed score of a play and also to represent the integer score displayed in the HUD).

File: osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs [hunk 2]
```diff
@@ -7,5 +7,76 @@ namespace osu.Game.Modes.Taiko.Judgements
...
+                    return 0;
+                case TaikoScoreResult.Good:
+                    return 100;
+                case TaikoScoreResult.Great:
+                    return 300;
+            }
+        }
+
+        /// <summary>
+        /// Computes the score value for the accurac portion of the score.
```
Comment (peppy) [inline osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs]: aurrac

File: osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs [hunk 3]
```diff
@@ -7,5 +7,76 @@ namespace osu.Game.Modes.Taiko.Judgements
...
+        /// </summary>
+        /// <param name="result">The result to compute the score value for.</param>
+        /// <returns>The int score value.</returns>
+        protected virtual int AccuracyScoreToInt(TaikoScoreResult result)
```
Comment (peppy) [inline osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs]: Maybe call these methods `NumericResultForScore` and `NumericResultForAccuracy`

File: osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs (hunk 1 above)
Comment (smoogipooo) [inline osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs]: `Result` conflicts with `JudgementInfo.Result` (type: HitResult). How about `ScoreResult`?

File: osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs [hunk 4]
```diff
@@ -10,71 +10,72 @@ public class TaikoJudgementInfo : JudgementInfo
         /// <summary>
         /// The maximum score value.
         /// </summary>
-        public const TaikoScoreResult MAX_SCORE = TaikoScoreResult.Great;
+        public const TaikoHitResult MAX_SCORE = TaikoHitResult.Great;
```
Comment (peppy) [inline osu.Game.Modes.Taiko/Judgements/TaikoJudgementInfo.cs]: MAX_HIT_RESULT

**Pull request merged**